apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js

```diff
@@ -418,4 +447,179 @@ describe('AllResidentBillingReceipts', () => {
+
+            // Prepare resident entities
+            const residentClient = await makeClientWithResidentUser()
+            const [ resident ] = await registerResidentByTestClient(residentClient, {
+                address: organizationClient.property.address,
+                addressMeta: organizationClient.property.addressMeta,
+                unitName: UNIT_NAME,
+            })
+
+            const [serviceConsumer] = await registerServiceConsumerByTestClient(residentClient, {
+                residentId: resident.id,
+                accountNumber: organizationClient.billingAccount.number,
+                organizationId: organizationClient.organization.id,
+            })
+            await registerServiceConsumerByTestClient(residentClient, {
+                residentId: resident.id,
+                accountNumber: billingAccount2.number,
+                organizationId: organizationClient.organization.id,
+            })
+
+            // Mobile app gets the list of all resident receipts
+            const [
+                singlePaymentReceiptBeforePayment,
+                noPaymentReceiptBeforePayment,
+            ] = await ResidentBillingReceipt.getAll(residentClient, { serviceConsumer: { resident: { id: resident.id } } }, { sortBy: ['toPay_ASC'] } )
+
+            // Mobile app tries to pay for the first receipt in one payment
+            await completeTestPayment(residentClient, admin, serviceConsumer.id, singlePaymentReceiptBeforePayment.id)
+
+            // Resident gets own receipts and sees that first one is fully paid!
+            const [
+                singlePaymentReceiptAfterPayment,
+                noPaymentReceiptAfterPayments,
+            ] = await ResidentBillingReceipt.getAll(residentClient, { serviceConsumer: { resident: { id: resident.id } } }, { sortBy: ['toPay_ASC'] })
+
+            // Assert with one payment for one receipt
+            expect(singlePaymentReceiptAfterPayment.id).toEqual(receiptWithSinglePayment.id)
+            expect(singlePaymentReceiptBeforePayment.id).toEqual(receiptWithSinglePayment.id)
+            expect(singlePaymentReceiptBeforePayment.paid).toEqual('0.00000000')
+            expect(singlePaymentReceiptAfterPayment.paid).toEqual('2000.00000000')
+            expect(singlePaymentReceiptAfterPayment.paid).toEqual(singlePaymentReceiptAfterPayment.toPay)
+
+            // Assert with no payments for one receipt
+            expect(noPaymentReceiptAfterPayments.id).toEqual(receiptWithNoPayments.id)
+            expect(noPaymentReceiptBeforePayment.id).toEqual(receiptWithNoPayments.id)
+            expect(noPaymentReceiptBeforePayment.paid).toEqual('0.00000000')
+            expect(noPaymentReceiptAfterPayments.paid).toEqual('0.00000000')
+            expect(noPaymentReceiptAfterPayments.paid).not.toEqual(noPaymentReceiptAfterPayments.toPay)
+        })
+
+        test('Test case when receipt is deleted then paid and then recreated.', async () => {
```

Comment:
```suggestion
        test('Test case when receipt is paid and then overwritten by billing integration.', async () => {
```

Reply:
It is not overwritten, but deleted and created new

Reply:
but definitely not `deleted then paid`